modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/RebalanceRaftGroupEventsListener.java

```diff
@@ -276,31 +265,53 @@ private void scheduleChangePeersAndLearners(PeersAndLearners peersAndLearners, l
     private void doStableKeySwitch(
             Set<Assignment> stableFromRaft,
             TablePartitionId tablePartitionId,
-            MetaStorageManager metaStorageMgr,
             long configurationTerm,
             long configurationIndex,
             BiFunction<TablePartitionId, Long, CompletableFuture<Set<Assignment>>> calculateAssignmentsFn
     ) {
-        try {
-            ByteArray pendingPartAssignmentsKey = pendingPartAssignmentsQueueKey(tablePartitionId);
-            ByteArray stablePartAssignmentsKey = stablePartAssignmentsKey(tablePartitionId);
-            ByteArray plannedPartAssignmentsKey = plannedPartAssignmentsKey(tablePartitionId);
-            ByteArray switchReduceKey = switchReduceKey(tablePartitionId);
-            ByteArray switchAppendKey = switchAppendKey(tablePartitionId);
-            ByteArray assignmentsChainKey = assignmentsChainKey(tablePartitionId);
-
-            // TODO: https://issues.apache.org/jira/browse/IGNITE-17592 Remove synchronous wait
-            Map<ByteArray, Entry> values = metaStorageMgr.getAll(
-                    Set.of(
-                            plannedPartAssignmentsKey,
-                            pendingPartAssignmentsKey,
-                            stablePartAssignmentsKey,
-                            switchReduceKey,
-                            switchAppendKey,
-                            assignmentsChainKey
-                    )
-            ).get();
+        doStableKeySwitchImpl(
```

Comment:
Is it for exception handling that you chain to doStableKeySwitchImpl in doStableKeySwitch? If yes, I'd rather rename doStableKeySwitchImpl to doStableKeySwitch and doStableKeySwitch to doStableKeySwitchWithExceptionHandling or similar.